src/main/java/uk/ac/cam/cl/dtg/isaac/dao/PgQuizAssignmentPersistenceManager.java

```diff
@@ -167,14 +167,17 @@ public Long saveAssignment(final QuizAssignmentDTO assignment) throws SegueDatab
     }
 
     @Override
-    public QuizAssignmentDTO getAssignmentById(Long quizAssignmentId) throws SegueDatabaseException {
+    public QuizAssignmentDTO getAssignmentById(Long quizAssignmentId) throws SegueDatabaseException, AssignmentCancelledException {
         try (Connection conn = database.getDatabaseConnection()) {
             PreparedStatement pst;
             pst = conn.prepareStatement("SELECT * FROM quiz_assignments WHERE id = ?");
             pst.setLong(1, quizAssignmentId);
 
             ResultSet results = pst.executeQuery();
             if (results.next()) {
+                if (results.getBoolean("deleted")) {
```

Comment:
Could this not just be an `AND NOT deleted` in the query, as we do for users and groups when they are deleted? Admittedly this is less useful information for the client, but perhaps saves the need for an exception?

Reply:
When James was explaining this to me I was going to suggest pushing the check and exception down as close to the DB as possible but it looks like this is already done. I personally am happy for the deleted check to be done at the persistence level and throwing a sensible checked exception if it improves the information to the user.

I guess one failure that I could see happening that we would like to avoid would be someone copying this method to create their own SELECT and not copying the deleted check. It is more practical than pretty... but we could get around that by adding a comment to the end of line 173 noting that deleted assignments are handled in code?

Reply:
Since this aims to cover exactly the same error case as happens with ordinary assignments, should we think about refactoring that @mlt47, or just start here first?

Reply:
Start here first, it is not Robin's responsibility anyway :)